File: widgetssdk/src/main/java/com/glia/widgets/view/floatingvisitorvideoview/FloatingVisitorVideoContract.java
package com.glia.widgets.view.floatingvisitorvideoview;

import com.glia.androidsdk.comms.MediaState;
import com.glia.widgets.base.BaseController;
import com.glia.widgets.base.BaseView;

public interface FloatingVisitorVideoContract {
    interface Controller extends BaseController {
        void setView(View view);
    }

    interface View extends BaseView<Controller> {
        void show(MediaState state);

        void hide();

        void onResume();

        void onPause();

        void showOnHold();

        void hideOnHold();
    }
}

File: widgetssdk/src/main/java/com/glia/widgets/view/floatingvisitorvideoview/FloatingVisitorVideoContract.kt
package com.glia.widgets.view.floatingvisitorvideoview

import com.glia.androidsdk.comms.MediaState

internal interface FloatingVisitorVideoContract {

    interface View {
        fun show(state: MediaState?)
        fun showFlipCameraButton(flipButtonState: FlipButtonState)
        fun hide()
        fun onResume()
        fun onPause()
        fun showOnHold()
        fun hideOnHold()
    }

    enum class FlipButtonState {
        SWITCH_TO_FACE_CAMERA,
        SWITCH_TO_BACK_CAMERA,
        HIDE
    }
}
